pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -622,11 +623,36 @@ func (ssc *defaultStatefulSetControl) rollingUpdateStatefulsetPods(
 		}
 	}
 
+	// handle specified deleted pod, and protected by maxUnavailable
+	for target := len(replicas) - 1; target >= 0; target-- {
+		if replicas[target] == nil {
+			continue
+		}
+		// the unavailable pods count exceed the maxUnavailable and the target is available, so we can't process it,
+		// why skip here rather than return?
+		// case: pod 0 ready, pod1 unready, pod 2 unready, pod3 ready, pod4 ready
+		// when maxUnavailable = 3, pod4 with specified deleted will be deleted but pod3 can't
+		// pod 2 and pod1 can be deleted because they were unavailable
+		if len(unavailablePods) >= maxUnavailable && !unavailablePods.Has(replicas[target].Name) {
+			continue
+		}
+		if specifieddelete.IsSpecifiedDelete(replicas[target]) {
+			if _, err := ssc.deletePod(set, replicas[target]); err != nil {
+				return status, err
+			}
+			// mark target as unavailable because it's updated
+			unavailablePods.Insert(replicas[target].Name)
+
+			if getPodRevision(replicas[target]) == currentRevision.Name {
+				status.CurrentReplicas--
```

Comment:
what if specified deleted pod is of new revision? shall we update status.UpdatedReplicas?